mantidproject/vesuvio
# Fixed analysis scripts, major clean-up of files

**Description of work:**
- Edited original scripts to take same inputs as current runner
- Fixed issue with Gamma Correction due to Mantid bug
- Ran original backward and forward scattering routines and checked that results match
- Removed unecessary files
- Included setting ip folder directory to the caching system through the CLI
- Organized some folders better

**To test:**
Currently the testing is a bit of a mess, but here is the procedure:

1. Install the mvesuvio package in the conda environment, I found that the quickest and easiest way to do this is through `pip install .` instead of using `conda build`, which takes longer.
2. Delete any previous caching files in `~/.mvesuvio`.
3. Run `mvesuvio config` and check caching happens for ip folder as well.
4. Create any experiment
5. Copy the attached python script for `analysis_inputs.py` into the created experiments folder.
6. Run `mvesuvio run`. You might have to press continue after loading the workspaces.
7. Run the python script `python Original_adapted_fwrd_script.py`.
8. Run the comparison script `python compare_current_with_original_results.py`. If everything ran without problems so far, the script will ask for an user input at the beginning. Press `y` to continue.
9. Check that in the plots that show up the blue and red dashed line match, should see the following:

![image](https://github.com/mantidproject/vesuvio/assets/80104863/558cac02-4232-4aa6-bd2c-8b288e6b4ccb)

Analysis Inputs script:
```python
import numpy as np


class LoadVesuvioBackParameters:
    def __init__(self, ipFilesPath):
        self.ipfile=ipFilesPath / "ip2019.par"
    runs="43066-43076"         # 77K         # The numbers of the runs to be analysed
    empty_runs="41876-41923"   # 77K         # The numbers of the empty runs to be subtracted
    spectra='3-134'                          # Spectra to be analysed
    mode='DoubleDifference'
    subEmptyFromRaw = True         # Flag to control wether empty ws gets subtracted from raw
    scaleEmpty = 1       # None or scaling factor
    scaleRaw = 1


class LoadVesuvioFrontParameters:
    def __init__(self, ipFilesPath):
        self.ipfile=ipFilesPath / "ip2018_3.par"
    runs='43066-43076'         # 100K        # The numbers of the runs to be analysed
    empty_runs='43868-43911'   # 100K        # The numbers of the empty runs to be subtracted
    spectra='144-182'                        # Spectra to be analysed
    mode='SingleDifference'
    subEmptyFromRaw = False         # Flag to control wether empty ws gets subtracted from raw
    scaleEmpty = 1       # None or scaling factor
    scaleRaw = 1


class GeneralInitialConditions:
    transmission_guess =  0.8537        # Experimental value from VesuvioTransmission
    multiple_scattering_order, number_of_events = 2, 1.e5
    vertical_width, horizontal_width, thickness = 0.1, 0.1, 0.001  # Expressed in meters


class BackwardInitialConditions(GeneralInitialConditions):
    def __init__(self, ipFilesPath):
        self.InstrParsPath=ipFilesPath / "ip2018_3.par"
    HToMassIdxRatio = 19.0620008206  # Set to zero or None when H is not present
    massIdx = 0
    masses = np.array([12, 16, 27])
    initPars = np.array([
            1, 12, 0.,
            1, 12, 0.,
            1, 12.5, 0.
        ])
    bounds = np.array([
            [0, np.nan], [8, 16], [-3, 1],
            [0, np.nan], [8, 16], [-3, 1],
            [0, np.nan], [11, 14], [-3, 1]
        ])
    constraints = ()
    noOfMSIterations = 3     #4
    firstSpec = 3    #3
    lastSpec = 134    #134
    maskedSpecAllNo = np.array([18, 34, 42, 43, 59, 60, 62, 118, 119, 133])
    MSCorrectionFlag = True
    GammaCorrectionFlag = False
    tofBinning='275.,1.,420'                    # Binning of ToF spectra
    maskTOFRange = None
    runHistData = True
    normVoigt = False


class ForwardInitialConditions(GeneralInitialConditions):
    def __init__(self, ipFilesPath):
        self.InstrParsPath = ipFilesPath / "ip2018_3.par"
    masses = np.array([1.0079, 12, 16, 27])
    initPars = np.array([
        1, 4.7, 0,
        1, 12.71, 0.,
        1, 8.76, 0.,
        1, 13.897, 0.
    ])
    bounds = np.array([
        [0, np.nan], [3, 6], [-3, 1],
        [0, np.nan], [12.71, 12.71], [-3, 1],
        [0, np.nan], [8.76, 8.76], [-3, 1],
        [0, np.nan], [13.897, 13.897], [-3, 1]
    ])
    constraints = ()
    noOfMSIterations = 3   #4
    firstSpec = 144   #144
    lastSpec = 182    #182
    MSCorrectionFlag = True
    GammaCorrectionFlag = True
    maskedSpecAllNo = np.array([173, 174, 179])
    tofBinning="110,1,430"                 # Binning of ToF spectra
    maskTOFRange = None
    runHistData = True
    normVoigt = False


class YSpaceFitInitialConditions:
    anything = True


class UserScriptControls:
    runRoutine = True
    procedure = "FORWARD"   # Options: None, "BACKWARD", "FORWARD", "JOINT"
    fitInYSpace = None    # Options: None, "BACKWARD", "FORWARD", "JOINT"


class BootstrapInitialConditions:
    runBootstrap = False

```
